app/controllers/ProductController.scala

```diff
@@ -99,35 +84,25 @@ class ProductController @Inject() (
    * (or an exception will get thrown) then check if it found a result
    * or not (result is zero) and return the appropriate status.
    *
-   * @param id
+   * @param id: String of the UUID given by the user in the URL
    * @return
    */
   def updateProduct(id: String): Action[JsValue] = Action(parse.json) { request =>
```

Comment:
```suggestion
  def updateProduct(id: String): Action[JsValue] = Action(parse.json[Products]) { request =>
```

should allow you to parse directly into the body parse and have `request.body` as a single element returned. As it stands, it is unclear what you really expect as a request body (a list? a single object?) whereas this needs to be very clear for your consumers

Reply:
As I do want to get a single object, the issue is if I parse it like that, I get a bad request like this
<img width="789" alt="Screen Shot 2022-11-23 at 12 40 55 PM" src="https://user-images.githubusercontent.com/60335436/203623722-14d4a868-aa18-459d-b446-47478790c805.png">
But if I change it to `Action(parse.json[List[Products]]` then it'll work though it will be a List of one product element inside it